samza-core/src/main/java/org/apache/samza/operators/impl/StreamTableJoinOperatorImpl.java

```diff
@@ -57,10 +57,10 @@ protected void handleInit(Config config, TaskContext context) {
   }
 
   @Override
-  public Collection<JM> handleMessage(KV<K, MV> message, MessageCollector collector, TaskCoordinator coordinator) {
-    K key = message.getKey();
-    RV recordValue = table.get(key);
-    KV<K, RV> record = recordValue != null ? KV.of(key, recordValue) : null;
+  public Collection<JM> handleMessage(M message, MessageCollector collector, TaskCoordinator coordinator) {
+    K key = ((KV<K, ?>) message).getKey();
```

Comment:
As a general join function implementation, I think we should invoke joinOpSpec.getMessageKeyFunction.apply(message), instead of class cast message to KV.